topological_navigation/scripts/localisation.py

```diff
@@ -161,6 +163,24 @@ def Callback(self, msg, item):
 
 
 
+    def get_nodes_wtag_cb(self,req):
+        rospy.wait_for_service('/topological_map_manager/get_tagged_nodes')
+        try:
+            cont = rospy.ServiceProxy('/topological_map_manager/get_tagged_nodes', strands_navigation_msgs.srv.GetTaggedNodes)
+            resp1 = cont(req.tag)
+            tagnodes = resp1.nodes
+        except rospy.ServiceException, e:
+            return "Service call failed: %s"%e
```

Comment:
That looks like terrible error handling... when there is an error you return a string, otherwise you return a list! I don't think this is a good idea...